kinoml/modeling/alignment.py

```diff
@@ -0,0 +1,206 @@
+        import numpy as np
+
+        # create the default matrix
+        # TODO add more options for the choice of matrix
+        matrix = align.SubstitutionMatrix.std_protein_matrix()
+
+        alignments = align.align_optimal(
+            seq.ProteinSequence(seq1),
+            seq.ProteinSequence(seq2),
+            matrix,
+            local=local,
+        )
+
+        alignment = alignments[0]
+
+        score = alignment.score
+        seq_identity = align.get_sequence_identity(alignment)
+        symbols = align.get_symbols(alignment)
+        codes = align.get_codes(alignment)
+
+        return cls(
+            alignment=alignment,
+            metadata={
+                "score": score,
+                "sequence_identity": seq_identity,
+                "symbols": symbols,
+                "codes": codes,
+            },
+        )
+
+    def _format_alignment_with_ligands(self, template, aligned_template_seq):
+
+        from MDAnalysis.lib.util import convert_aa_code
+
+        sel = template.universe.select_atoms("protein and name CA")
+        resnames = sel.resnames.tolist()
+        resnames_one_letter = [convert_aa_code(x) for x in resnames]
+
+        # get the last four AAs in the aligned target sequence
+        end_chunk = aligned_template_seq[-4:][::-1]
+        aa_store = []
+        # loop over residues in reverse
+        for i, letter in enumerate(resnames_one_letter[::-1]):
+            # read a chunk of four AAs
+            current_chunk = resnames_one_letter[::-1][i : i + 4]
+
+            # if the end of the aligned target sequence is hit, stop
+            if current_chunk == end_chunk:
+                break
+            # otherwise store the one letter AAs that need to be added
+            else:
+                aa_store.append(letter)
+
+        # create a one element list of dashes
+        dashes = "".join(["-" for x in aa_store])
+        # create one element list of AAs in the canonical order
+        amino_acids = "".join(aa_store[::-1])
+
+        return amino_acids, dashes
+
+    def make_ali_file(
+        self,
+        aligned_template_seq: str,
+        aligned_target_seq: str,
+        template: ProteinStructure,
+        target: Union[str, KinaseDomainAminoAcidSequence],
+        ligand: bool = False,
+    ):
+        """
+        Generate an alignment file in MODELLER format
+        ----------
+        aligned_template_seq: str
+            The aligned template sequence
+        aligned_target_seq: str
+            The aligned target sequence
+        template: ProteinStructure
+            The template structure
+        target: list of str or KinaseDomainAminoAcidSequence
+            The original target sequence
+        ligand: bool
+            Specify whether to retain a ligand in the alignment.
+            (Default: False)
+        Returns
+        -------
+        """
+
+        # Convert None entries into dashes
+        conv = lambda i: i or "-"
```

Comment:
I don't think you need a `lambda` here (and named lambdas are not usually recommended, since it's preferred to do `def conv(i): i or "-"`). In this case, you can embed the expression in the list comprehension below:

```python
 seq1_dashed = [i or "-" for i in aligned_template_seq]
```